java/maven/src/org/netbeans/modules/maven/execute/model/NetbeansActionMapping.java

```diff
@@ -391,4 +392,71 @@ public String getModelEncoding()
     {
         return modelEncoding;
     }
+
+    @Override
+    public int hashCode() {
+        int hash = 7;
+        hash = 89 * hash + Objects.hashCode(this.actionName);
+        hash = 89 * hash + Objects.hashCode(this.displayName);
+        hash = 89 * hash + (this.recursive ? 1 : 0);
+        hash = 89 * hash + Objects.hashCode(this.packagings);
+        hash = 89 * hash + Objects.hashCode(this.goals);
+        hash = 89 * hash + Objects.hashCode(this.properties);
+        hash = 89 * hash + Objects.hashCode(this.options);
+        hash = 89 * hash + Objects.hashCode(this.activatedProfiles);
+        hash = 89 * hash + Objects.hashCode(this.basedir);
+        hash = 89 * hash + Objects.hashCode(this.preAction);
+        hash = 89 * hash + Objects.hashCode(this.reactor);
+        hash = 89 * hash + Objects.hashCode(this.modelEncoding);
+        return hash;
```

Comment:
suggest https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Objects.html#hash(java.lang.Object...)

Reply:
I did let NB generate those two methods since me writing this by hand has always the risk that I overlook something. At some point we should improve the code gen for newer APIs.

re: its exported friend API and the POJO is mutable atm, not sure if it is worth converting.